src/types/geometrycollection.rs

```diff
@@ -84,6 +86,27 @@ where
     }
 }
 
+impl<T: WktNum> GeometryCollectionTrait for GeometryCollection<T> {
+    type T = T;
+    type GeometryType<'a> = &'a Wkt<T> where Self: 'a;
+
+    fn dim(&self) -> geo_traits::Dimensions {
+        if self.0.is_empty() {
```

Comment:
Can this be parsed from the input?

`GEOMETRYCOLLECTION ZM EMPTY`

That might be a larger change though - requiring explicit tracking of the dimensionality of the geometry.

Reply:
I didn't recall that empty geometries could also have a dimension. Does simple features allow that?

Reply:
I guess I'm not sure. I just assumed that this was valid:

`GEOMETRYCOLLECTION Z(POLYGON Z EMPTY, POINT Z(1 2 3))`

Maybe it's not?

Reply:
Yeah you're right, it is valid: 
<img width="484" alt="image" src="https://github.com/user-attachments/assets/81bc4922-b7a0-467a-8b9a-4fc829457a92">
(I assume shapely/geos implement Simple Features accurately)

Reply:
I'd say that yes, long-term we should figure out a way to maintain the dimension for empty WKT geometries.